src/aead/block.rs

```diff
@@ -79,7 +79,8 @@ impl BitXor for Block {
 impl From<&'_ [u8; BLOCK_LEN]> for Block {
     #[inline]
     fn from(bytes: &[u8; BLOCK_LEN]) -> Self {
-        Self(FromByteArray::from_byte_array(bytes))
+        let bytes: &[[u8; BLOCK_LEN / 2]; 2] = bytes.chunks_fixed();
+        Self(bytes.map(Into::into))
```

Comment:
This is ugly, but at the end of this little project to remove `unsafe` from `ring::endian` this will change again to something obvious.